.drone.yml

```diff
@@ -85,29 +85,63 @@ steps:
+    privileged: true
     depends_on:
       - core
       - principal
+    volumes:
+      - name: sock
+        path: /var/run/docker.sock
     commands:
       - git clone https://github.com/enigmampc/discovery-docker-network.git
       - cd discovery-docker-network && cp .env-template .env
       - sed -i "s/COMPOSE_PROJECT_NAME=.*/COMPOSE_PROJECT_NAME=enigma_${DRONE_BUILD_NUMBER}/" .env
-      - if [[ ${DRONE_BRANCH} == "master" ]]; then export TAG=latest; else export TAG=develop; fi
-      - if [[ $TAG == "develop" ]]; then sed -i "s/DOCKER_TAG=latest/DOCKER_TAG=develop/" .env; fi
-      - docker-compose pull
+      - export MATCHING_BRANCH_P2P="$(git ls-remote --heads https://github.com/enigmampc/enigma-p2p.git ${DRONE_BRANCH} | wc -l)"
+      - export MATCHING_BRANCH_CONTRACT="$(git ls-remote --heads https://github.com/enigmampc/enigma-contract.git ${DRONE_BRANCH} | wc -l)"
+      - export DOCKER_TAG=core_${DRONE_BUILD_NUMBER}
+      - sed -i "s/DOCKER_TAG=latest/DOCKER_TAG=${DOCKER_TAG}/" .env;
+      - |
+        if [[ "${DRONE_BRANCH}" == "master" ]]; then
+          docker pull enigmampc/enigma_contract:latest
+          docker pull enigmampc/enigma_p2p:latest
+          docker tag enigmampc/enigma_contract:latest enigmampc/enigma_contract:$DOCKER_TAG
+          docker tag enigmampc/enigma_p2p:latest enigmampc/enigma_p2p:$DOCKER_TAG
+        else
+          if [ $MATCHING_BRANCH_P2P -eq 0 ] && [ $MATCHING_BRANCH_CONTRACT -eq 0 ]; then
+            docker pull enigmampc/enigma_contract:develop
+            docker pull enigmampc/enigma_p2p:develop
+            docker tag enigmampc/enigma_contract:develop enigmampc/enigma_contract:$DOCKER_TAG
+            docker tag enigmampc/enigma_p2p:develop enigmampc/enigma_p2p:$DOCKER_TAG
+          else
+            if [ $MATCHING_BRANCH_P2P -eq 1 ] && [ $MATCHING_BRANCH_CONTRACT -eq 0 ]; then
+              cd enigma-p2p && docker build --build-arg GIT_BRANCH_P2P=${DRONE_BRANCH} -t enigmampc/enigma_p2p:$DOCKER_TAG --no-cache . && cd ..
+              docker pull enigmampc/enigma_contract:develop
+              docker tag enigmampc/enigma_contract:develop enigmampc/enigma_contract:$DOCKER_TAG
+            else
+              if [ $MATCHING_BRANCH_P2P -eq 0 ] && [ $MATCHING_BRANCH_CONTRACT -eq 1 ]; then
+                cd enigma-contract && docker build --build-arg GIT_BRANCH_CONTRACT=${DRONE_BRANCH} -t enigmampc/enigma_contract:$DOCKER_TAG --no-cache . && cd ..
+                docker pull enigmampc/enigma_p2p:develop
+                docker tag enigmampc/enigma_p2p:develop enigmampc/enigma_p2p:$DOCKER_TAG
+              else
+                cd enigma-p2p && docker build --build-arg GIT_BRANCH_P2P=${DRONE_BRANCH} -t enigmampc/enigma_p2p:$DOCKER_TAG --no-cache . && cd ..
+                cd enigma-contract && docker build --build-arg GIT_BRANCH_CONTRACT=${DRONE_BRANCH} -t enigmampc/enigma_contract:$DOCKER_TAG --no-cache . && cd ..
+              fi
+            fi
+          fi
+        fi
```

Comment:
This should be more maintainable:
```
export MATCHING_BRANCH_P2P="$(git ls-remote --heads https://github.com/enigmampc/enigma-p2p.git ${DRONE_BRANCH} | wc -l)"
export MATCHING_BRANCH_CONTRACT="$(git ls-remote --heads https://github.com/enigmampc/enigma-contract.git ${DRONE_BRANCH} | wc -l)"
export DOCKER_TAG=core_${DRONE_BUILD_NUMBER}
sed -i "s/DOCKER_TAG=latest/DOCKER_TAG=${DOCKER_TAG}/" .env;

declare -a PROJECTS=(p2p contract)

declare -A DOCKER_IMAGES=(
    [p2p]=enigma_p2p
    [contract]=enigma_contract
)

declare -A GIT_BRANCH_ARG=(
    [p2p]=GIT_BRANCH_P2P
    [contract]=GIT_BRANCH_CONTRACT
)

declare -A PROJECT_BRANCH_FOUND=(
    [p2p]=MATCHING_BRANCH_P2P
    [contract]=MATCHING_BRANCH_CONTRACT
)

declare -A PROJECT_DIRECTORY=(
    [p2p]=enigma-p2p
    [contract]=enigma-contract
)

for project in "${PROJECTS}"; do
    DOCKER_IMAGE="enigmampc/${DOCKER_IMAGES[$project]}"
    if [[ "$DRONE_BRANCH" == "master" ]]; then
        docker pull "$DOCKER_IMAGE"
        docker tag "$DOCKER_IMAGE:latest" "$DOCKER_IMAGE:$DOCKER_TAG"
    # The ! mark here is "indirect expansion"
    elif [ ${!PROJECT_BRANCH_FOUND[$project]} -eq 0 ]; then
        docker pull "$DOCKER_IMAGE"
        docker tag "$DOCKER_IMAGE:develop" "$DOCKER_IMAGE:$DOCKER_TAG"
    elif [ ${!PROJECT_BRANCH_FOUND[$project]} -eq 1 ]; then
        cd ${PROJECT_DIRECTORY[$project]} && docker build --build-arg ${GIT_BRANCH_ARG[$project]}=${DRONE_BRANCH} -t "$DOCKER_IMAGE:$DOCKER_TAG" --no-cache . && cd ..
    fi
done
```

Reply:
Oh... i just realized you can remove the indirection trick in the `elif` clause if you expand the `MATCHING_BRANCH_*` parts in its definition (just prefix a `$`), but i'll keep it there so you can see that it's possible if you need it :+1:

Reply:
Thanks @reuvenpo for the suggestion. After much trial and error, this has been added in #5917a2f